addons/purchase/models/product.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.addons.base.models.res_partner import WARNING_MESSAGE, WARNING_HELP
+
+
+class ProductTemplate(models.Model):
+    _name = 'product.template'
+    _inherit = 'product.template'
+
+    @api.multi
+    def _purchase_count(self):
+        for template in self:
+            template.purchase_count = sum([p.purchase_count for p in template.product_variant_ids])
+        return True
+
+    property_account_creditor_price_difference = fields.Many2one(
+        'account.account', string="Price Difference Account", company_dependent=True,
+        help="This account will be used to value price difference between purchase price and cost price.")
+    purchase_count = fields.Integer(compute='_purchase_count', string='# Purchases')
+    purchase_method = fields.Selection([
+        ('purchase', 'On ordered quantities'),
+        ('receive', 'On received quantities'),
+    ], string="Control Policy", help="On ordered quantities: control bills based on ordered quantities.\n"
+        "On received quantities: control bills based on received quantity.", default="receive")
+    purchase_line_warn = fields.Selection(WARNING_MESSAGE, 'Purchase Order Line', help=WARNING_HELP, required=True, default="no-message")
+    purchase_line_warn_msg = fields.Text('Message for Purchase Order Line')
+
+
+class ProductProduct(models.Model):
+    _name = 'product.product'
+    _inherit = 'product.product'
+
+    @api.multi
+    def _purchase_count(self):
+        domain = [
+            ('state', 'in', ['purchase', 'done']),
+            ('product_id', 'in', self.mapped('id')),
+        ]
+        PurchaseOrderLines = self.env['purchase.order.line'].search(domain)
```

Comment:
Please find an improvement fot this compute : https://github.com/odoo/odoo/pull/23101